pkg/ingester/client/client.go

```diff
@@ -36,11 +39,13 @@ type closableHealthAndIngesterClient struct {
 // MakeIngesterClient makes a new IngesterClient
 func MakeIngesterClient(inst ring.InstanceDesc, cfg Config, metrics *Metrics, logger log.Logger) (HealthAndIngesterClient, error) {
 	logger = log.With(logger, "component", "ingester-client")
-	var reportGRPCStatusesOptions []middleware.InstrumentationOption
-	if cfg.ReportGRPCStatusCodes {
-		reportGRPCStatusesOptions = []middleware.InstrumentationOption{middleware.ReportGRPCStatusOption}
+	unary, stream := grpcclient.Instrument(metrics.requestDuration, middleware.ReportGRPCStatusOption)
+	// cfg.DeprecatedReportGRPCStatusCodes is deprecated.
+	// Starting from Mimir 2.14.0, Mimir behavior should be as this flag were set to true.
+	// TODO: Remove the following check in Mimir 2.14.0
+	if !cfg.DeprecatedReportGRPCStatusCodes {
+		unary, stream = grpcclient.Instrument(metrics.requestDuration)
```

Comment:
[nit] I would prefer to keep previous code logic. By resetting `unary` and `stream` vars here, it's too easy to introduce bug into the code by inserting more wrapping between lines 42 and 46. (Is there unit test covering this? If so, then likelihood of introducing such bug is low)

Reply:
Agree with Peter. I think it's the same exact feedback I already gave you on another PR (probably a dskit one).

Reply:
Thank you both. I will revert this change.